webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/editablemapping/EditableMappingIOService.scala

```diff
@@ -0,0 +1,146 @@
+package com.scalableminds.webknossos.tracingstore.tracings.editablemapping
+
+import com.scalableminds.util.io.{NamedFunctionStream, ZipIO}
+import com.scalableminds.util.time.Instant
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.webknossos.datastore.datareaders.{
+  BloscCompressor,
+  IntCompressionSetting,
+  StringCompressionSetting
+}
+import com.scalableminds.webknossos.datastore.datareaders.zarr3.{
+  BloscCodecConfiguration,
+  BytesCodecConfiguration,
+  ChunkGridConfiguration,
+  ChunkGridSpecification,
+  ChunkKeyEncoding,
+  ChunkKeyEncodingConfiguration,
+  EmptyZarr3GroupHeader,
+  Zarr3ArrayHeader,
+  Zarr3DataType
+}
+import com.scalableminds.webknossos.tracingstore.files.TsTempFileService
+import com.typesafe.scalalogging.LazyLogging
+import jakarta.inject.Inject
+
+import java.io.{BufferedOutputStream, File, FileOutputStream}
+import java.nio.ByteBuffer
+import java.nio.file.Path
+import java.util.zip.Deflater
+import scala.concurrent.ExecutionContext
+
+class EditableMappingIOService @Inject()(tempFileService: TsTempFileService) extends LazyLogging {
+
+  // 10000 edges per chunk (an edge is two Longs in edges and one bool in edgeIsAddition)
+  private val ChunkSize: Int = 10000
+  private val arrayNameEdges = "edges"
+  private val arrayNameEdgeIsAddition = "edgeIsAddition"
+
+  def editedMappingEdgesToZippedZarrTempFile(editedEdges: Seq[(Long, Long, Boolean)], tracingId: String)(
+      implicit ec: ExecutionContext): Fox[Path] =
+    for {
+      before <- Instant.nowFox
+      edgesZarrChunks: Iterator[Array[Byte]] = editedEdgesToZarrChunks(editedEdges)
+      isAdditionZarrChunks: Iterator[Array[Byte]] = edgeIsAdditionToZarrChunks(editedEdges)
+      edgesZarrChunksStream = edgesZarrChunks.zipWithIndex.map {
```

Comment:
Isn't is more like `edgesZarrChunkStreams` as it is an iterator of `NamedFunctionStream`? 
Same for `isAdditionZarrChunksStream` -> `isAdditionZarrChunkStreams`